src/sentry/uptime/detectors/rdap.py

```diff
@@ -0,0 +1,149 @@
+import ipaddress
+from collections.abc import Mapping, MutableMapping, Sequence
+from socket import gethostbyname
+from typing import Any, TypedDict
+
+import requests
+from django.core.cache import cache
+
+RDAP_BOOTSTRAP_REGSITRY = "https://data.iana.org/rdap/ipv4.json"
+"""
+JSON service registry maintained by the IANA containing a mapping of ipv4 CIDRs
+to RDAP service urls.
+"""
+
+CIDRList = Sequence[str]
```

Comment:
Should this be a set instead for faster lookups?

Reply:
nah we have to iterate the list to find the match anyway. There's not many either so I'm not worried